src/main/java/io/xpring/ilp/DefaultIlpClient.java

```diff
@@ -0,0 +1,165 @@
+package io.xpring.ilp;
+
+import org.interledger.spsp.server.grpc.AccountServiceGrpc;
+import org.interledger.spsp.server.grpc.BalanceServiceGrpc;
+import org.interledger.spsp.server.grpc.CreateAccountRequest;
+import org.interledger.spsp.server.grpc.CreateAccountResponse;
+import org.interledger.spsp.server.grpc.GetAccountRequest;
+import org.interledger.spsp.server.grpc.GetAccountResponse;
+import org.interledger.spsp.server.grpc.GetBalanceRequest;
+import org.interledger.spsp.server.grpc.GetBalanceResponse;
+import org.interledger.spsp.server.grpc.IlpOverHttpServiceGrpc;
+import org.interledger.spsp.server.grpc.SendPaymentRequest;
+import org.interledger.spsp.server.grpc.SendPaymentResponse;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.StatusRuntimeException;
+import io.xpring.ilp.grpc.IlpJwtCallCredentials;
+import io.xpring.xrpl.XpringException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Optional;
+import java.util.Properties;
+
+/**
+ * A client that can create accounts, get accounts, get balances, and send ILP payments on a connector.
+ */
+public class DefaultIlpClient implements IlpClientDecorator {
+
+    private static String XPRING_ILP_GRPC_URL;
+    private static final Logger logger = LoggerFactory.getLogger(DefaultIlpClient.class);
+
+    // Bring in configuration properties such as GRPC url
+    static {
+        String propertiesFileName = "client.properties";
+        Properties properties = new Properties();
+
+        try {
+            // Make sure that the configuration file exists
+            URL res = DefaultIlpClient.class.getClassLoader().getResource(propertiesFileName);
+            InputStream inputStream = new FileInputStream(res.getFile());
+            // load the properties file
+            properties.load(inputStream);
+
+            XPRING_ILP_GRPC_URL = properties.getProperty("ilp.grpc.url");
+        } catch (Exception e) {
+            logger.info(String.format("Problem loading config file named %s. Using default configuration.", propertiesFileName));
+            XPRING_ILP_GRPC_URL = "http://localhost:6565";
+        }
+    };
+
+    private final BalanceServiceGrpc.BalanceServiceBlockingStub balanceServiceStub;
```

Comment:
This has been a point of confusion for @theotherian and me. It's unclear what GRPC best practices are. I remember in one of our initial meetings about GRPC, you said that there should be a 1 to 1 between protobuf messages and GRPC endpoints (aka CreateAccountRequest, GetAccountRequest even though they return the same fields).  

Putting all of your service methods into one monolithic service seems like an anti-pattern...

Is there somewhere where gRPC best practices are documented?

Reply:
@keefertaylor I have a question around this as it's come up in previous PRs around integrating our stuff.

I'm not sure I understand why this is considered a desirable direction/best practice? Is there a best practices document/guide somewhere that outlines this more clearly? It seems like condensing all of your services into a single file would end up, at the very least, resulting in a mess of disparate methods that don't fit in one place.

Reply:
Oh, I meant that if you had some message that was like:

```
message Foo {}

message GetFooRequest{}
message GetBarRequest{}
```

then it would be weird to have something like:
```
service BarService {
   rpc foo(getFooRequest) returns (Foo)
   rpc foo(getBarRequest) returns (Foo)
}
```

I can't find a logical explanation of best practice in [buf.build]() or [prototool]() or [Google Cloud's style guide]().

I would look at it as a `service` corresponds a single server API. So I would think that you'd just define `HermesService` with 3 rpcs (get balance, send, make account).

If we ever got to the point where there were two binaries running, say we split the accounts service out into xpring-eng/HermesAccountService and we still have Hermes then I would support two gRPC definitions. 

If you use multiple services, I don't think you can bind them all to the same port right? So you would run balanceService on :5001, accountService on :5002, and sendService on :5003. This is confusing for clients who need to input these URLs. 
In general, I assume that the server handling for gRPC is going to be using resources on a per server basis (if nothing else, open sockets and threads in your binary). 

I am happy to let this go in and I think it's generally a trivial refactor. I'm sorry for being unclear about this.

Reply:
Upon looking at this though you've managed to use one RPC port and presumably your integration tests pass so maybe I am confused about how listeners work

Reply:
All good, I think pretty much everyone is confused about how listeners work/best practices.  

I'm probably going to refactor the messages a bit, but I'll do that in a later PR.

Reply:
> If you use multiple services, I don't think you can bind them all to the same port right? So you would run balanceService on :5001, accountService on :5002, and sendService on :5003. This is confusing for clients who need to input these URLs.
In general, I assume that the server handling for gRPC is going to be using resources on a per server basis (if nothing else, open sockets and threads in your binary).

As far as what we've seen, they bind to the same port without a problem, and I'm not sure why it would cause a problem because the name of the service is part of the URL if I recall correctly. Given that the service is scoped at the resource level, it doesn't seem like you would run into any kind of unpredictable collisions. 

> I would look at it as a service corresponds a single server API. So I would think that you'd just define HermesService with 3 rpcs (get balance, send, make account).

Perhaps this is just a difference in thinking with regard to how a RESTful service would look. For example, in REST you wouldn't put all of your resources under a single path (outside of using a path like `/admin` or `/private` for some kind of permission scoping in the URL), you'd usually have something like `/products`, `/users`, etc. Since it seems like a gRPC service loosely acts as a 1 to 1 with what a RESTful resource would be, in our minds it made sense to separate the two.

On that note, looking at some of the resources you passed along, in this one I would infer that there's at least a loose suggestion that having multiple services is a reasonable thing to do: https://buf.build/docs/lint-checkers#service_suffix

It looks like others have attempted the same in terms of running multiple services on the same port; sounds like it's just a matter of attaching multiple to the same server instance:
https://stackoverflow.com/questions/52634217/access-multiple-grpc-services-over-the-same-connection